### PR TITLE
Update Guava dependency version to 32.0.1-jre

### PR DESCRIPTION
**Pull request by volodk85**
Upgrade guava dependency to 32.0.1 or higher

Gradle verification metadata was updated using command:
```
./gradlew --write-verification-metadata sha256
```

### CODE REVIEW

**Comment by volodk85** [comment]
Jenkins run elasticsearch-ci/bwc

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -664,6 +664,11 @@
             <sha256 value="a42edc9cab792e39fe39bb94f3fca655ed157ff87a8af78e1d6ba5b07c4a00ab" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.google.guava" name="guava" version="32.0.1-jre">
```
Comment (kingherc) [inline gradle/verification-metadata.xml:667]: nit. I'm a bit concerned why we keep also the hashes of the old versions, e.g., why we keep the hashes of the previous 31.1-jre version exactly above? I think we could remove them.

**Review by kingherc — APPROVED** [review]
LGTM

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (volodk85) [reply gradle/verification-metadata.xml:667]: If I remove them manually and re-run gradle tooling, those dependencies come back - surely they're transitive for some other libraries

**Pull request merged**